src/main/java/it/pagopa/pn/downtime/consumer/LegalFactIdReceiver.java

```diff
@@ -34,26 +40,38 @@ public class LegalFactIdReceiver {
 	LegalFactService legalFactService;
 
 	@SqsListener(value = "${amazon.sqs.end-point.legalfact-available}", deletionPolicy = SqsMessageDeletionPolicy.ON_SUCCESS)
-	public void receiveLegalFact(final String message) throws Exception {
+	public void receiveLegalFact(final String message) throws JsonProcessingException {
 		log.info("threadId : {}, currentTime : {}", Thread.currentThread().getId(), System.currentTimeMillis());
-		log.info("message received in Legal Facts queue {}", message);	
-		FileCreatedDto legalFact = mapper.readValue(message, FileCreatedDto.class);
-		
-		Map<String, AttributeValue> eav1 = new HashMap<>();
-		eav1.put(":legalFact1", new AttributeValue().withS(legalFact.getKey()));
-		DynamoDBScanExpression scanExpression = new DynamoDBScanExpression()
-			    .withFilterExpression("legalFactId =:legalFact1")
-			    .withExpressionAttributeValues(eav1);
-		
-		List<DowntimeLogs> logs = dynamoDBMapper.scanPage(DowntimeLogs.class, scanExpression).getResults();
+		PnAuditLogBuilder auditLogBuilder = new PnAuditLogBuilder();
+		PnAuditLogEvent logEvent = auditLogBuilder.before(PnAuditLogEventType.AUD_NT_DOWNTIME,
+				"message received in Legal Facts queue {}", message)
+			.build();
+		logEvent.log();
+		try {
+			FileCreatedDto legalFact = mapper.readValue(message, FileCreatedDto.class);
+
+			Map<String, AttributeValue> eav1 = new HashMap<>();
+			eav1.put(":legalFact1", new AttributeValue().withS(legalFact.getKey()));
+			DynamoDBScanExpression scanExpression = new DynamoDBScanExpression()
+					.withFilterExpression("legalFactId =:legalFact1").withExpressionAttributeValues(eav1);
+
+			List<DowntimeLogs> logs = dynamoDBMapper.scanPage(DowntimeLogs.class, scanExpression).getResults();
+
+			updateFileAvailable(logs, legalFact);
+			logEvent.generateSuccess().log();
+		} catch (ResourceNotFoundException exc) {
+			logEvent.generateFailure(exc.getMessage()).log();
+			log.error("STACKTRACE: {}", ExceptionUtils.getStackTrace(exc));
```

Comment:
Gli audit log non devono riportare lo stack trace, sono log che vengono archiviati per 10 anni e hanno valore nelle indagini a livello legale. Devono essere chiari senza entrare nei dettaglio, se è neccessario lo stack trace va messo un log a parte.

Reply:
Preferisci lo porti a livello di debug o che lo tolga?

Reply:
Se non viene trovata alcuna corrispondenza, nel metodo updateFileAvailable viene sollevata l'eccezione ResourceNotFoundException con un messaggio che descrive l'errore e indica che non è stato trovato alcun record nel db.  
In seguito nel catch viene poi passato all'audit log (logEvent) il messaggio dell'eccezione stessa. 
Ho aggiunto lo stack trace per agevolare eventualmente il troubleshooting, ma può essere rimosso. Lo devo rimuovere e loggare solo il messaggio custom o lascio lo stacktrace?

Reply:
Se il problema è un record non trovato sul DB posso mettere nel log di audit che non ho trovato quel record indicando i criterio di ricerca.
Il caso in questione è proprio un errore che deve sollevare un allarme e buttare giù dal letto il reperibile di turno o è un errore gestito dal codice?

Reply:
Certo, se non viene trovata alcuna corrispondenza sul DB viene sollevata l'eccezione a riga 74: **throw new ResourceNotFoundException("No Downtime Found for legalFactId {} = " + legalFact.getKey());**.
In seguito a riga 63 nel catch attraverso **logEvent.generateFailure(exc.getMessage()).log();** viene passato al log di audit il messaggio dell'eccezione lanciata a riga 74, ovvero _"No Downtime Found for legalFactId {} = " + legalFact.getKey()_. Quindi, se non viene trovata corrispondenza, nel log di audit verrà segnalato che è stato trovato quel record con il relativo criterio di ricerca. Invece il log a riga 64 **log.error("STACKTRACE: {}", ExceptionUtils.getStackTrace(exc));** non è relativo al log di audit, ma è un log che apparirà unicamente in console ed è stato inserito per agevolare il troubleshooting. Quest'ultimo si potrebbe rilvelare utile nel caso di errori o bug, ma può essere rimosso. Lo devo rimuovere e loggare solo il messaggio custom o lascio lo stacktrace?

Il caso in questione è un errore che difficilmente può capitare e, nel caso limite in cui dovesse accadere, l'applicazione lo gestisce non andando ad interferire con le esecuzioni per i rimanenti legalFactId su coda.